src/cpp/redis.cpp

```diff
@@ -101,7 +80,7 @@ CommandReply Redis::run(CommandList& cmds)
     CommandList::iterator cmd_end = cmds.end();
     CommandReply reply;
     while(cmd != cmd_end) {
-        reply = this->run(**cmd);
+        reply = ((Command *)(*cmd))->runme(this);
```

Comment:
We've decided to move away from c-style casts to c++ casts, so we should change ``(Command *)`` to the appropriate cast.  I also wonder if a private ``Redis`` function like ``Redis::_run_base_command(Command& cmd)`` would make this easier so that all of the methods could just call that without having to do the pointer creation and dereferencing seen in functions above.